feat(lite): implement useContext composable

ByScripts opened the pull request:
### Description

This PR introduces a new `useContext` composable.

This allows to pass down a value to any children at any depth, with the ability for a child to update the value along the chain.

```ts
// context.ts

export const DisabledContext = createContext(false);
```

```ts
// SomeParentComponent.vue

const isDisabled = useContext(DisabledContext) // Will return the default context value: false

const isDisabled = useContext(DisabledContext, true) // Update and return the context value

const isDisabled = useContext(DisabledContext, () => props.disabled) // Value can also be updated from props (see `Caveats` in doc about `boolean` props)
```

```ts
// SomeChildAtAnyDepth.vue

const isDisabled = useContext(DisabledContext); // Will get the value set by the parent component

const isDisabled = useContext(DisabledContext, () => props.disabled) // Will override the value set by the parent component and update the context for subsequent children.
```

```ts
// SiblingComponent.vue

const isDisabled = useContext(DisabledContext); // Sibling component will not be concerned by updated context and get the default value.
```

It's also possible to pass a context builder as second parameter to `createContext`:

```ts
const CounterContext = createContext(10, (count) => ({
  count,
  double: computed(() => count.value * 2)
}))

const { count, double } = useContext(CounterContext);
```
### Checklist

- Commit
  - Title follows [commit conventions](https://bit.ly/commit-conventions)
  - Reference the relevant issue (`Fixes #007`, `See xoa-support#42`, `See https://...`)
  - If bug fix, add `Introduced by`
- Changelog
  - If visible by XOA users, add changelog entry
  - Update "Packages to release" in `CHANGELOG.unreleased.md`
- PR
  - If UI changes, add screenshots
  - If not finished or not tested, open as _Draft_

ByScripts committed:
feat(lite): new context system

MathieuRA commented on @xen-orchestra/lite/src/components/ui/UiCard.vue:
```diff
@@ -1,24 +1,40 @@
+
+const { name: contextColor, backgroundClass } = useColorContext(
+  () => props.color
+);
```
it sound weird to have `ColorContext` and also `useColorContext`.

MathieuRA commented on @xen-orchestra/lite/src/composables/context.composable.ts:
```diff
@@ -0,0 +1,36 @@
+    key: Symbol() as InjectionKey<ComputedRef<T>>,
+    defaultValue,
+  };
+};
```
I think it may be nice to have the possibility to create a more "complex" context, to avoid creating a wrapper as was the case for `ColorContext` and `useColorContext`

Do you think something like this would be possible?
```js
export const createContext = (factory, ...args) => {
  const defaultValue =
    typeof factory === "function" ? factory(...args) : factory;

  return {
    key: Symbol(),
    defaultValue,
  };
};
```
And instead of creating two contexts for the color, do something like:

```js
export const ColorContext = createContext(
  (color: Color) => ({
    value: color,
    bgClass: computed(() => `foobar-${color}`),
    textClass: computed(() => `foobar-text-${color}`),
  }),
  "info"
);

const { value: color, bgClass, textClass } = useContext(ColorContext)

ByScripts committed:
feat(lite/context): feedback

Removed useColorContext in favor of a new Context builder allowing to customize the context output
